### PR TITLE
Develop upstream sync 230612

### PR DESCRIPTION
**Pull request by draganmladjenovic**
https://github.com/ROCmSoftwarePlatform/frameworks-internal/issues/4664

### CODE REVIEW

**Comment by draganmladjenovic** [comment]
retest Ubuntu-CPU please

**Comment by jayfurmanek** [comment]
Missed a merge conflict at: `tensorflow/compiler/tests/xla_call_module_test.py", line 372`

**Comment by i-chaochen** [comment]
//tensorflow/compiler/tests:xla_call_module_test_gpu
//tensorflow/compiler/xla/service/gpu/tests:gpu_kernel_tiling_test

these two are failed in both GPU-single and gpu-pycpp

**Comment by jayfurmanek** [comment]
retest cpu-pycpp please

File: tensorflow/compiler/tests/xla_call_module_test.py [hunk 1]
```diff
@@ -321,13 +321,9 @@ def f(x):
                              Sout=[()],
                              platforms=platforms)
 
-<<<<<<< HEAD
-    expected_value = x + dict(CPU=2., CUDA=3., ROCM=4., TPU=4.)[self.testing_platform()]
-=======
     expected_value = (
-        x + dict(CPU=2.0, CUDA=3.0, ROCM=3.0, TPU=4.0)[self.testing_platform()]
+        x + dict(CPU=2.0, CUDA=3.0, ROCM=4.0, TPU=4.0)[self.testing_platform()]
```
Comment (i-chaochen) [inline tensorflow/compiler/tests/xla_call_module_test.py:325]: @draganmladjenovic  Wonder is xla_call_module_test failed because of ROCM=3.0 ?

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-multi please
retest Ubuntu-GPU-single please
retest code-check please
retest cpy-pycpp please
reset gpu-non-pip-multi please
reset gpu-pycpp please

**Comment by i-chaochen** [comment]
retest cpu-pycpp please
reset gpu-non-pip-multi please
reset gpu-pycpp please

**Comment by i-chaochen** [comment]
reset gpu-pycpp please
reset gpu-non-pip-multi please

**Comment by draganmladjenovic** [comment]
retest Ubuntu-GPU-single please
retest gpu-non-pip-multi please
retest gpu-pycpp please

**Comment by draganmladjenovic** [comment]
retest Ubuntu-GPU-single please

**Comment by draganmladjenovic** [comment]
retest Ubuntu-GPU-single please

**Comment by draganmladjenovic** [comment]
retest Ubuntu-GPU-single please

**Comment by draganmladjenovic** [comment]
retest Ubuntu-GPU-single please

**Comment by draganmladjenovic** [comment]
retest Ubuntu-GPU-single please

**Comment by draganmladjenovic** [comment]
retest Ubuntu-GPU-single please

**Review by i-chaochen — APPROVED** [review]
Wonder is xla_call_module_test failed because of ROCM=3.0 ?

**Review by i-chaochen — COMMENTED** [review]
Please upstream this commit @draganmladjenovic  thanks!

**Pull request merged**